astropy/nddata/utils.py

```diff
@@ -519,9 +519,10 @@ def __init__(self, data, position, size, wcs=None, mode='trim',
             units of pixels.  ``size`` can also be a
             `~astropy.units.Quantity` object or contain
             `~astropy.units.Quantity` objects.  Such
-            `~astropy.units.Quantity` objects must be in pixel or angular
-            units.  See the ``mode`` keyword for additional details on
-            the final cutout size.
+            `~astropy.units.Quantity` objects must be in pixel or
+            angular units.  If ``size`` contains any floats, they will
```

Comment:
The rounding of angular sizes was already in there:
https://github.com/astropy/astropy/pull/4216/files#diff-9d3c0ca15cfa1e9110d749db5d607202R649
Now, `size` is always rounded.

BTW, I came across this issue testing `size = 0.3*u.arcsec / (0.1*u.arsec/u.pix)  [= 2.999999.... pix]`, expecting `3x3` pixel cutouts, but getting `2x2` cutouts.